internal/export/ingest_pipelines.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package export
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+	"github.com/elastic/elastic-package/internal/elasticsearch/ingest"
+)
+
+type PipelineWriteLocationType string
+
+const (
+	PipelineWriteLocationTypeDataStream PipelineWriteLocationType = "data_stream"
+	PipelineWriteLocationTypeRoot       PipelineWriteLocationType = "root"
+)
+
+// Represents a target write location for exporting an ingest pipeline
+type PipelineWriteLocation struct {
+	Type       PipelineWriteLocationType
+	Name       string
+	ParentPath string
+}
+
+func (p PipelineWriteLocation) WritePath() string {
+	return filepath.Join(p.ParentPath, "elasticsearch", "ingest_pipeline")
+}
+
+type PipelineWriteAssignments map[string]PipelineWriteLocation
+
+func IngestPipelines(ctx context.Context, api *elasticsearch.API, writeAssignments PipelineWriteAssignments) error {
+	var pipelineIDs []string
+
+	for pipelineID := range writeAssignments {
+		pipelineIDs = append(pipelineIDs, pipelineID)
+	}
+
+	pipelines, err := ingest.GetRemotePipelinesWithNested(ctx, api, pipelineIDs...)
+
+	if err != nil {
+		return fmt.Errorf("exporting ingest pipelines using Elasticsearch failed: %w", err)
+	}
+
+	pipelineLookup := make(map[string]ingest.RemotePipeline)
+	for _, pipeline := range pipelines {
+		pipelineLookup[pipeline.Name()] = pipeline
+	}
+
+	err = writePipelinesToFiles(writeAssignments, pipelineLookup)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writePipelinesToFiles(writeAssignments PipelineWriteAssignments, pipelineLookup map[string]ingest.RemotePipeline) error {
+	if len(writeAssignments) == 0 {
+		return nil
+	}
+
+	for name, writeLocation := range writeAssignments {
+		pipeline, ok := pipelineLookup[name]
+		if !ok {
+			continue
+		}
+		err := writePipelineToFile(pipeline, writeLocation)
+		if err != nil {
+			return err
+		}
+
+		depPipelineWriteAssignments := createWriteAssignments(writeLocation, pipeline.GetProcessorPipelineNames())
+		err = writePipelinesToFiles(depPipelineWriteAssignments, pipelineLookup)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func writePipelineToFile(pipeline ingest.RemotePipeline, writeLocation PipelineWriteLocation) error {
+	var jsonPipeline map[string]any
+	err := json.Unmarshal(pipeline.JSON(), &jsonPipeline)
+	if err != nil {
+		return fmt.Errorf("unmarshalling ingest pipeline failed (ID: %s): %w", pipeline.Name(), err)
+	}
+
+	yamlBytes, err := yaml.Marshal(jsonPipeline)
```

Comment:
I wonder if we should do some formatting here. Some things:
* Remove `_meta`, specially the `managed*` fields should not be needed here.
* Remove `version`, I think this is not being used in packages.
* Indentation. Looking to pipelines they use to be indented by two spaces, default marshalling seems to use four spaces.

Maybe we should make `elastic-package format` to automatically format these files too. But that is another story.

Reply:
I didn't come across cases where the `version` field showed up in a pipeline while exporting, I assume you are seeing this as a top-level field in some instances?

Reply:
Yes, it appears at the root level. After your latest changes it doesn't appear anymore. Thanks.